tests/components/enphase_envoy/test_switch.py

```diff
@@ -109,7 +111,32 @@ async def test_switch_grid_operation(
     mock_envoy.go_off_grid.assert_awaited_once_with()
     mock_envoy.go_off_grid.reset_mock()
 
-    test_entity = f"{Platform.SWITCH}.enpower_{sn}_charge_from_grid"
+
+@pytest.mark.parametrize(
+    ("mock_envoy", "use_envoy_serial"),
+    [
+        ("envoy_metered_batt_relay", False),
+        ("envoy_eu_batt", True),
+    ],
+    indirect=["mock_envoy"],
+)
+async def test_switch_charge_from_grid_operation(
+    hass: HomeAssistant,
+    mock_envoy: AsyncMock,
+    config_entry: MockConfigEntry,
+    use_envoy_serial: bool,
+) -> None:
+    """Test switch platform operation for charge from grid switches."""
+    with patch("homeassistant.components.enphase_envoy.PLATFORMS", [Platform.SWITCH]):
+        await setup_integration(hass, config_entry)
+
+    sn = (
+        f"envoy_{mock_envoy.serial_number}"
+        if use_envoy_serial
+        else f"enpower_{mock_envoy.data.enpower.serial_number}"
+    )
```

Comment:
We should remove the if statement, so rather just put the serial numbers as parameter.I think

Reply:
> We should remove the if statement, so rather just put the serial numbers as parameter.I think

These are now removed.